homeassistant/components/rainbird.py

```diff
@@ -0,0 +1,105 @@
+"""
+Module for interacting with WiFi LNK module of the Rainbird Irrigation system.
+
+This project has no affiliation with Rainbird. This module works with the
+Rainbird LNK WiFi Module. For more information see:
+http://www.rainbird.com/landscape/products/controllers/LNK-WiFi.htm
+
+This module communicates directly towards the IP Address of the WiFi module it
+does not support the cloud. You can start/stop the irrigation and get the
+currenltly active zone.
+
+I'm not a Python developer, so sorry for the bad code. I've developed it to
+control it from my domtica systems.
+"""
+
+import logging
+import homeassistant.helpers as helpers
+
+REQUIREMENTS = ['pyrainbird==0.0.7']
+
+# Home Assistant Setup
+DOMAIN = 'rainbird'
+SERVER = ''
+PASSWORD = ''
+STATE_VAR = 'rainbird.activestation'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """
+    Call from Home Assistant.
+
+    @param hass: default homeassistant hass class
+    @param config: default homeassistant config class
+    """
+    server = config[DOMAIN].get('stickip')
+    password = config[DOMAIN].get('password')
+
+    # RainbirdSetup
+    from pyrainbird import RainbirdController
+
+    controller = RainbirdController(_LOGGER)
+    controller.setConfig(server, password)

```

Comment:
How do you ensure that the password is correct? What happens if the device is not accessible?

Reply:
There are 2 failure scenario's. 0 is a NOT ACK from the controller, -1 is a failure (timeout, config etc after 1 retry). Debug level logging should give an indication towards the problem.

Reply:
At the moment wouldn't the users end-up with a non-functional platform in their setup if their password is wrong? The users would only be able to recognize this problem after calling the service.

Reply:
Better to raise `PlatformNotReady` exception and setup will be re-tried after Home Assistant has launched (only available once migrated to a switch platform)